src/osd/OSD.cc

```diff
@@ -2903,7 +2903,9 @@ will start to track new ops received afterwards.";
   }
 
   else if (prefix == "heap") {
-    ret = ceph::osd_cmds::heap(*cct, cmdmap, *f, ss);
+    std::stringstream outss;
+    ret = ceph::osd_cmds::heap(*cct, cmdmap, outss, ss);
```

Comment:
this does not compile.

Reply:
Mixed investigating, cleaning and fixing altogether. Apologizes.